packages/pyright-internal/src/analyzer/program.ts

```diff
@@ -1259,19 +1259,26 @@ export class Program {
                         !invokedFromUserFile ||
                         this._isUserCode(curSourceFileInfo)
                     ) {
-                        this._bindFile(curSourceFileInfo);
+                        this._parseFile(curSourceFileInfo);
```

Comment:
its okay but you might even skip parsing if text doesn't have text you are looking for?

Reply:
ex) https://github.com/microsoft/pyright/blob/fdd1e3b85feb8693a2e95ce37d762cb53ef3a310/packages/pyright-internal/src/analyzer/program.ts#L1933

Reply:
The cost of tokenizing and parsing is small compared to the cost of reading in the file contents the first time. And once it's parsed, we retain the file contents as part of the parse results, so it's faster on subsequent searches.

Reply:
I stand corrected. I implemented your suggested change. It didn't have that much of an effect on subsequent searches (3.3s vs 3.2s), but it greatly sped up the first search. From 16s to 6.8s. So I think that's a good optimization.